File: Lab_7/Main.java

import java.util.Scanner;

//Clase base
class Empleado{
    private String nombre;
    private String id;
    private double salarioBase;
    
    public Empleado(String nombre, String id, double salarioBase){
        this.nombre = nombre;
        this.id = id;
        this.salarioBase = salarioBase;
    }
    
    public String getNombre(){
        return nombre;
    }
    
    public void setNombre(String nombre){
        this.nombre = nombre;
    }
    
    public String getId(){
        return id;
    }
    
    public void setId(String id){
        this.id = id;
    }
    
    public double getSalarioBase(){
        return salarioBase;
    }
    
    public void setSalarioBase(double salarioBase){
        this.salarioBase = salarioBase;
    }
    
    public double calcularSalario(){
        return salarioBase;
    }
}

//Clase derivada de Empleado
class EmpleadoTiempoCompleto extends Empleado{
    public EmpleadoTiempoCompleto(String nombre, String id, double salarioBase){
        super(nombre, id, salarioBase);
    }
    
    @Override
    public double calcularSalario(){
        return getSalarioBase() * 1.10;
    }
} 

//Clase derivada de Empleado
class EmpleadoMedioTempo extends Empleado{
    public EmpleadoMedioTempo(String nombre, String id, double salarioBase) {
        super(nombre, id, salarioBase);
    }
    
    @Override
    public double calcularSalario(){
        return getSalarioBase() * 0.50;
    }
}

//Clase derivada de Empleado
class EmpleadoPorHoras extends Empleado{
    private int horasTrabajadas;
    private static final double TARIFA_POR_HORA = 15.0;
    
    public EmpleadoPorHoras(String nombre, String id, double salarioBase) {
        super(nombre, id, salarioBase);
    }
    
    public void setHorasTrabajadas(int horas){
        this.horasTrabajadas = horas;
    }
    
    @Override
    public double calcularSalario(){
        return horasTrabajadas * TARIFA_POR_HORA;
    }
}

//Clase principal
public class Main{
    public static void main(String[] args){
        Scanner lector = new Scanner(System.in);
        
        System.out.println("Ingresa el nombre del empleado de tiempo completo: ");
        String nombre = lector.nextLine();
        System.out.println("Ingresa su id: ");
        String id = lector.nextLine();
        System.out.println("Ingresa su salario base: ");
        double salario = lector.nextDouble();
        lector.nextLine();
        Empleado empleado1 = new EmpleadoTiempoCompleto(nombre, id, salario);
        
        
        System.out.println("\nIngresa el nombre del empleado de medio tiempo: ");
        nombre = lector.nextLine();
        System.out.println("Ingresa su id: ");
        id = lector.nextLine();
        System.out.println("Ingresa su salario base: ");
        salario = lector.nextDouble();
        lector.nextLine();
        Empleado empleado2 = new EmpleadoMedioTempo(nombre, id, salario);
        
        
        System.out.println("\nIngresa el nombre del empleado por horas: ");
        nombre = lector.nextLine();
        System.out.println("Ingresa su id: ");
        id = lector.nextLine();
        System.out.println("Ingresa sus horas trabajadas: ");
        int horas = lector.nextInt();
        EmpleadoPorHoras empleado3 = new EmpleadoPorHoras(nombre, id, 0);
        empleado3.setHorasTrabajadas(horas);
        
        lector.close();

        System.out.println("\nSalario de " + empleado1.getNombre() + ": $" + empleado1.calcularSalario());
        System.out.println("Salario de " + empleado2.getNombre() + ": $" + empleado2.calcularSalario());
        System.out.println("Salario de " + empleado3.getNombre() + ": $" + empleado3.calcularSalario());
    }
}

File: Lab_7/Main.py
#Clase base
class Empleado:
    def __init__(self, nombre, id, salarioBase):
        self._nombre = nombre
        self._id = id
        self._salarioBase = salarioBase

    @property
    def nombre(self):
        return self._nombre
    
    @nombre.setter
    def nombre(self, nombre):
        self._nombre = nombre

    @property
    def id(self):
        return self._id
    
    @id.setter
    def id(self, id):
        self._id = id

    @property
    def salarioBase(self):
        return self._salarioBase
    
    @salarioBase.setter
    def salarioBase(self, salarioBase):
        self._salarioBase = salarioBase

    def calcular_salario(self):
        return self._salarioBase
    
#Clase derivada de Empleado
class EmpleadoTiempoCompleto(Empleado):
    def calcular_salario(self):
        return self.salarioBase * 1.10
    
#Clase derivada de Empleado
class EmpleadoMedioTempo(Empleado):
    def calcular_salario(self):
        return self.salarioBase * 0.50
    
#Clase derivada de Empleado
class EmpleadoPorHoras(Empleado):
    TARIFA_POR_HORA = 15.0
    
    def __init__(self, nombre, id, salarioBase=0):
        super().__init__(nombre, id, salarioBase)
        self.horasTrabajadas = 0

    def set_horasTrabajadas(self, horas):
        self.horasTrabajadas = horas

    def calcular_salario(self):
        return self.horasTrabajadas * self.TARIFA_POR_HORA
    
#Programa principal
if __name__ == "__main__":
    nombre = input("Ingresa el nombre del empleado de tiempo completo: ")
    id = input("Ingresa su id: ")
    salario = float(input("Ingresa su salario base: "))
    empleado1 = EmpleadoTiempoCompleto(nombre, id, salario)

    nombre = input("Ingresa el nombre del empleado de medio tiempo: ")
    id = input("Ingresa su id: ")
    salario = float(input("Ingresa su salario base: "))
    empleado2 = EmpleadoMedioTempo(nombre, id, salario)

    nombre = input("Ingresa el nombre del empleado por horas: ")
    id = input("Ingresa su id: ")
    horas = int(input("Ingresa sus horas trabajadas: "))
    empleado3 = EmpleadoPorHoras(nombre, id, 0)
    empleado3.set_horasTrabajadas(horas)

    print(f"\nSalario de {empleado1.nombre}: ${empleado1.calcular_salario():.2f}")
    print(f"Salario de {empleado2.nombre}: ${empleado2.calcular_salario():.2f}")
    print(f"Salario de {empleado3.nombre}: ${empleado3.calcular_salario():.2f}")